### PR TITLE
Add PhoneInput component

### PR DESCRIPTION
**Pull request by luciatoma**
# Description

Created new Phone Details component

## Changes

- [x] New phoneDetailsInput component

## How to test

- Checkout this branch
- `$ npm run dev`

### CODE REVIEW

File: src/components/Select/index.js [hunk 1]
```diff
@@ -106,6 +119,11 @@ export function Select({ items, floatingMenu, onChange, id, label, ...props }) {
             }}
             aria-selected={selectedItem === item ? 'true' : 'false'}
           >
+            {flag && (
+              <StyledFlagsContainer>
+                <Flag countryCode={item ? item.value : null} />
+              </StyledFlagsContainer>
+            )}
```
Comment (rppld) [inline src/components/Select/index.js:122]: I see you added the flag as `leftAdornment`-prop above, so I think this block can be removed?

Comment (rppld) [reply src/components/Select/index.js:122]: Actually I think all references to `flag` can be removed from the `<Select />` component.

File: src/components/PhoneInput/index.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+            label="Country"
+            hideLabel
+            items={options}
+            floatingMenu
+            validate={validateCountry}
+            initialSelectedItem={options[intialSelectedIndex]}
+            leftAdornment={
+              <Flag countryCode={country ? country.value : 'nl'} />
+            }
+            flag
```
Comment (rppld) [inline src/components/PhoneInput/index.js]: Can be removed I think?

File: src/components/PhoneInput/index.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+  initialSelectedCountry,
+  label,
+  onChange,
+  validateCountry,
+  validateNumber,
+  ...props
+}) {
+  const inputRef = React.useRef()
+  const options = createSelectOptions(countries)
+
```
Comment (rppld) [inline src/components/PhoneInput/index.js]: Empty line.

File: src/components/PhoneInput/index.js [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+  const intialSelectedIndex = initialSelectedCountry
+    ? countries.indexOf(initialSelectedCountry)
+    : 0
+  const [country, setCountry] = React.useState(countries[intialSelectedIndex])
+
```
Comment (rppld) [inline src/components/PhoneInput/index.js:79]: Empty line.

**Review by rppld — COMMENTED** [review]
Few small comments. Tested it out and works really well 👍

**Comment by rppld** [comment]
On iOS, I like that the `<Select />` doesn’t have the flag to save space! But do you think it‘s somehow possible to show the country-name next to the country-code when the dropdown is open (the list at the bottom of the screenshot)?

Something like: `Netherlands (+31)`

IMG_65032099C946-1

**Review by lucasfeliciano — APPROVED** [review]

File: src/components/PhoneInput/index.js [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+          <Select
+            id="country"
+            label="Country"
+            hideLabel
+            items={options}
+            floatingMenu
+            validate={validateCountry}
+            initialSelectedItem={options[intialSelectedIndex]}
+            leftAdornment={
+              <Flag countryCode={country ? country.value : 'nl'} />
```
Comment (RobVermeer) [inline src/components/PhoneInput/index.js]: I don't think we should have default stuff like this in Solar.

**Comment by RobVermeer** [comment]
@rppld @luciatoma Agreed, the select box should open the names on mobile. Just like we currently have in the sell flow. If you don't know your country digits, you also don't know them while scrolling through the list 😅 

Example of current implementation in the sell flow.
Screenshot 2019-06-24 14 27 25

File: src/components/PhoneInput/index.stories.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  },
+]
+
+storiesOf('PhoneInput', module)
+  .add('basic', () => (
+    <PhoneDetailsInput
+      id="phonenumber"
+      label="Phone number"
+      countries={countries}
+      onChange={e => console.log(e)}
```
Comment (RobVermeer) [inline src/components/PhoneInput/index.stories.js:39]: Just to make it more clear in the stories, use both vars here with proper names.

`onChange={(country, number) => console.log(country, number)}` for example.

When testing I thought it was a bug that the number wasn't showing after change, but on reading the code I saw it was the second param (=

**Comment by RobVermeer** [comment]
Screenshot 2019-06-24 16 29 39

React warning in console when trying the mobile one.

**Comment by RobVermeer** [comment]
Screenshot 2019-06-24 16 30 23

And another warning for the ref id.

**Review by RobVermeer — APPROVED** [review]
Works as expected 👌

**Pull request merged**